connector-builder-server/src/main/openapi/openapi.yaml

```diff
@@ -0,0 +1,217 @@
+          application/json:
+            schema:
+              $ref: "#/components/schemas/StreamsListRequestBody"
+        required: true
+      responses:
+        "200":
+          description: Successful operation
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/StreamsListRead"
+        "400":
+          $ref: "#/components/responses/ExceptionResponse"
+        "422":
+          $ref: "#/components/responses/InvalidInputResponse"
+
+components:
+  schemas:
+    StreamRead:
+      type: object
+      required:
+        - results
+      properties:
+        results:
+          type: array
+          description: The RECORD and STATE AirbyteMessages coming from the read operation
+          items:
+            type: string
+        logs:
+          type: array
+          description: The LOG AirbyteMessages coming from the read operation
+          items:
+            type: string
+        httpRequests:
+          type: array
+          description: HTTP requests sent to the source API, and their responses
+          items:
+            type: object
+            required:
+              - request
+              - response
+            properties:
+              request:
+                type: object
+                description: The HTTP request sent to the source API
+                required:
+                  - url
+                  - headers
+                properties:
+                  url:
+                    type: string
+                    description: URL that the request was sent to
+                  body:
+                    type: object
+                    description: The body of the HTTP request, if present
+                  headers:
+                    type: object
+                    description: The headers of the HTTP request
+              response:
+                type: object
+                description: The response from the source API
+                required:
+                  - status
+                  - headers
+                properties:
+                  status:
+                    type: integer
+                    description: The status of the response
+                  body:
+                    type: object
+                    description: The body of the HTTP response, if present
+                  headers:
+                    type: object
+                    description: The headers of the HTTP response
+    StreamReadRequestBody:
+      type: object
+      required:
+        - definition
+        - stream
+      properties:
+        definition:
+          $ref: "#/components/schemas/ConnectorDefinitionBody"
+          description: The config-based connector definition contents
+        stream:
+          type: string
+          description: Name of the stream to read
+    ConnectorDefinitionBody:
+      $ref: ../../../../airbyte-cdk/python/airbyte_cdk/sources/declarative/config_component_schema.json
```

Comment:
I chose here to use the actual config_component_schema JSON schema to define the request body structure of these endpoints, with the idea that the YAML editor contents can be converted into a JSON object and sent to the server in that form. This way, the OpenAPI spec is communicating what the structure of the file should be, which may allow us to do some better input validations/autocomplete/etc on the frontend side.

The alternative is to just make this a string field, but that feels too loose and does not give us the above benefit.

This relative path here is a little gross though. That could be solved by moving this openapi.yaml file into `airbyte-cdk`. Open to any other suggestions here

Reply:
I like that we don't have to redefine everything here